### DISCUSSION TITLE
Multi-partitions support for more than 2 partitions?

### DISCUSSION THREAD

**Discussion by DreamwareDevelopment** [dagster-io/dagster#20495] [Ideas]
My company needs multiple static partitions along with a time based partition. Are there any plans for expanding the allowed number of partitions in a multi partition definition? Is there a workaround? I could just iterate the third set of static non-partition partitions in the asset, however, as that set grows, backfilling becomes expensive because I have to backfill every partition that executed prior to the date of the newest non-partition partition being added even though N - 1 of the non-partition partitions have already been executed.

**Comment by garethbrickman** [comment]
I think this may fall under the ambit of this issue: https://github.com/dagster-io/dagster/issues/14228 it's worth documenting your use case in its comments to collate!

**Reply by DreamwareDevelopment** [reply]
Sorry for the late response. It wasn't critical then but it's becoming a critical need soon. I need two static partitions and one date partition. One static for city, another for permit type, then a dynamic for date. However, now that I'm writing this I think I may be able to just work with concatenating strings for the static partition keys. It's not optimal, as I'd like to run all the permit types under one city easily with the UI but it should work just fine to concatenate.

**Comment by alexismanin** [comment]
I've got the same need. I've got a pipeline that fetch datasets and publish them. I would like to be able to fetch, publish and track state of publicaiton using 3 criteria  :

 - administrative area
 - producer of the data
 - date of production

Ideally, we 'd like to be able to select any combination of the three criteria, and track state of computation using partition status.
Therefore, I would like to be able to define partitions like this : 

```python
daily_partitions = DailyPartitionsDefinition(start_date="2020-01-01")
zone_partitions = StaticPartitionsDefinition(["AREA-1", "AREA-2", "AREA-3", ...])
producer_partitions = StaticPartitionsDefinition(["LAB-1", "PROD-1", "PARTNER-1", ...])
publishing_partitions = MultiPartitionsDefinition(
    {
        "date": daily_partitions,
        "producer": producer_partitions,
        "zone": zone_partitions,
    }
)
```

and then fetch partition keys like that : 

```python
# partition_key looks like "2024-01-01|aperture laboratories|us"
partition_key: MultiPartitionKey = publishing_partitions.get_partition_key_from_str(context.partition_key)
pkey = partition_key.keys_by_dimension
date = datetime.date.fromisoformat(pkey["date"])
producer = pkey["producer"]
zone = pkey["zone"]
```

But, for now, I have to trick dagster by merging manually both static partitions, which cause the following problems:

 * Must merge and split partition keys manually
 * Tracking individual dimensions state becomes more difficult.

For now, I therefore define partitions as follow : 

```python
daily_partitions = DailyPartitionsDefinition(start_date="2020-01-01")
producers = ["LAB-1", "PROD-1", "PARTNER-1", ...]
zones = ["AREA-1", "AREA-2", "AREA-3", ...]
producer_x_zone_list = [ " / ".join((producer, zone)) for producer in producers for zone in zones]
producer_x_zone_partitions = StaticPartitionsDefinition(producer_x_zone_list)
publishing_partitions = MultiPartitionsDefinition(
    {
        "date": daily_partitions,
        "producer / zone": producer_x_zone_partitions,
    }
)
```

and then get back individual keys/dimensions like so : 

```python
# partition_key looks like "2024-01-01|aperture laboratories / us"
partition_key: MultiPartitionKey = publishing_partitions.get_partition_key_from_str(context.partition_key)
pkey = partition_key.keys_by_dimension
date = datetime.date.fromisoformat(pkey["date"])
producer_x_zone = pkey["producer / zone"]
producer, zone = producer_x_zone.rsplit(" / ", maxsplit=1)
```

**Comment by zander-dav** [comment]
Bump on this. We are still limited to just 2 partitions definitions when using `MultiPartitionsDefinition`, forcing many users to adopt clunky string parsing in asset definitions
